cql3/statements/create_view_statement.cc

```diff
@@ -260,6 +260,25 @@ view_ptr create_view_statement::prepare_view(data_dictionary::database db) const
         throw exceptions::invalid_request_exception(format("Must select at least a column for a Materialized View"));
     }
 
+    // Cassandra requires that if CLUSTERING ORDER BY is used, it must specify
+    // all clustering columns. Scylla relaxes this requirement and allows just
+    // a subset of the clustering columns. But it doesn't make sense (and it's
+    // forbidden) to list something which is not a clustering key column in
+    // the CLUSTERING ORDER BY. Let's verify that:
+    for (auto& pair: _properties.defined_ordering()) {
+        auto&& name = pair.first->text();
+        bool not_clustering = true;
+        for (auto& c : _clustering_keys) {
+            if (name == c->to_string()) {
+                not_clustering = false;
+                break;
+            }
+        }
+        if (not_clustering) {
+            throw exceptions::invalid_request_exception(format("CLUSTERING ORDER BY lists {} which is not a clustering column in the view", name));
+        }
+    }
+
```

Comment:
> Maybe it makes sense, though, to commit this patch first and then think about making it even more strict?

I guess we can live with more lenient checks for now to preserve compatibility.

I think there is one more check that should be added to make sure that the ORDERING BY is correct.
The new verification doesn't check if the same column appears multiple times.
Because of that doing something like this works, even though it should be rejected:
```cql
... WITH CLUSTERING ORDER BY (pk ASC, pk DESC);
```

Reply:
I'll add such a test as an xfail test for https://github.com/scylladb/scylladb/issues/12936.
But I think we should start by committing this patch. It aims to solve exactly one problem, issue https://github.com/scylladb/scylladb/issues/10767, and doesn't solve any other issues. I want to solve the other issues in the context of #12936.

Reply:
I have a draft pull request https://github.com/scylladb/scylladb/issues/10767 to add more tests after this PR is committed.
@scylladb/scylla-maint please consider merging this PR.

Reply:
Queued.

Reply:
Added a test (actually just part of a test) to check duplicate clustering key columns are also not accepted to https://github.com/scylladb/scylladb/pull/12938.